File: Ad-Hoc/pagina02/1196/1196.py
teclado = [
    [" ", "1", "2", "3", "4", "5", "6", "7", "8", "9", "0", "-", "="],
    ["Q", "W", "E", "R", "T", "Y", "U", "I", "O", "P", "[", "]", "\\"],
    ["A", "S", "D", "F", "G", "H", "J", "K", "L", ":", "´"],
    ["Z", "X", "C", "V", "B", "N", "M", ",", ".", "/"]
]

mapeamento = {}
for linha in teclado:
    for i in range(1, len(linha)):
        mapeamento[linha[i]] = linha[i - 1]

cache = {}

while True:
    try:
        frase = input()
        frase_corrigida = ""

        for char in frase:
            if char == " ":
                frase_corrigida += " "
            else:
                if char not in cache:
                    cache[char] = mapeamento.get(char, "")
                frase_corrigida += cache[char]
        print(frase_corrigida)
    except EOFError:
        break

File: Ad-Hoc/__1000__/1196/1196.cs
using System;
using System.Collections.Generics;

class URI {

    static void Main() {
        string teclado = " ` 1234567890-=\nQWERTYUIOP[]\\\nASDFGHJKL;'\nZXCVBNM,./";

        var mapeamento = new Dictionary<char, char>();
        foreach (var linha in teclado.Split('\n')){
            for(int i = 1; i < linha.Length; i++){
                mapeamento[linha[i]] = linha[i - 1];
            }
        }

        string input;
        while((input = Console.ReadLine()) != null){

            string resultado = "";
            foreach(char c in input){
                if(c == " ")
                    resultado += ' ';
                else if(mapeamento.ContainsKey(c))
                    resultado += mapeamento[c];
                else
                    resultado += c;
            }
            Console.WriteLine(resultado);
        }
    }
}